src/util.js

```diff
@@ -44,7 +44,7 @@ export const recursiveMerge = (
     // TODO(jlfwong): Handle malformed input where a and b are not the same
     // type.
 
-    if (typeof a !== 'object') {
+    if (typeof a !== 'object' || Array.isArray(b)) {
```

Comment:
Do we maybe want to check if `a` is an array as well? We want merging `{a: [1]}` and `{a: 2}` to do something reasonable here.